### ISSUE TITLE
Add SetCookie prefixing feature for `__Secure-` and `__Host-`

### ISSUE THREAD

**Issue by Manouchehri** [honojs/hono#1203]
https://www.sjoerdlangkemper.nl/2017/02/09/cookie-prefixes/

So for `__Secure-`, the usage would look like:

```typescript
setCookie(c, 'example', 'secret_example', {
  path: '/api/demo',
  httpOnly: true,
  sameSite: 'Strict',
  securePrefix: true,
  secure: true // if omitted, set to true. If set to false, throw an error
})
```

This would result in the cookie name actually being set to `__Secure-example`.

```typescript
const incoming_cookie = getCookie(c, 'example', {
  securePrefix: true
})
```

And for `__Host-`, the usage would look like:

```typescript
setCookie(c, 'example', 'secret_example', {
  path: '/', // if omitted, set to /. If set to anything other than /, then throw an error.
  httpOnly: true,
  sameSite: 'Strict',
  hostPrefix: true,
  secure: true // if omitted, set to true. If set to false, throw an error
  // If domain is set, throw an error
})
```

```typescript
const incoming_cookie = getCookie(c, 'example', {
  hostPrefix: true
})
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#cookie_prefixes

**Comment by yusukebe** [comment]
Hi @Manouchehri 

Ah, this is a good feature! Will you implement this?

**Comment by Manouchehri** [comment]
Not in the immediate future, wasn't 100% sure if you would agree with the error handling/throws. 😄

**Comment by yusukebe** [comment]
This was closed by #2269. Thank you.
